### PR TITLE
Run tests on every push to a base branch

### PR DESCRIPTION
**Pull request by poncovka**
We need to update the coverage of these branches in the codecov.io,
so we can compare the pull request coverage with the base coverage.

**Tested on:** https://github.com/poncovka/anaconda/runs/2686386124

### CODE REVIEW

**Review by VladimirSlavik — APPROVED** [review]
LGTM, I agree that this should not use the environment.

Thanks for fixing the coverage thing!

**Review by rvykydal — APPROVED** [review]
Looks good to me.

File: .github/workflows/push-tests.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+jobs:
+  unit-tests:
+    runs-on: ubuntu-20.04
+    timeout-minutes: 30
+    steps:
+      - name: Clone repository
+        uses: actions/checkout@v2
+
+      - name: Build anaconda-ci container
+        run: make -f Makefile.am anaconda-ci-build
```
Comment (jkonecny12) [inline .github/workflows/push-tests.yml:18]: Do not build the container, just let it download from our quay.io repository. These will be downloaded automatically on when starting the tests. So just removing these two lines should be enough.

Or re-use this part https://github.com/rhinstaller/anaconda/blob/master/.github/workflows/tests.yml#L57 .

**Review by jkonecny12 — APPROVED** [review]
Looks good to me. However, I don't think we should build the containers.

**Review by jkonecny12 — CHANGES_REQUESTED** [review]
Sorry it should have been NACK :D

**Review by jkonecny12 — APPROVED** [review]
After off-site discussion there is a valid reason to use this way. We are not able to detect changed files so in case there will be dependency change or Dockerfile change we won't be able to detect that.
On the other hand we are not guarded against the changes from the system. So, if some dependency will break Anaconda we would get it here but not from the repository because it's gated before the upload.

Removing NACK, this is about to choose the smaller evil in this case.

**Pull request merged**